src/realm/util/string_buffer.hpp

```diff
@@ -132,16 +130,11 @@ inline const char* StringBuffer::data() const noexcept
     return m_buffer.data();
 }
 
-inline char* StringBuffer::c_str() noexcept
-{
-    char* d = data();
-    return d ? d : &m_zero;
-}
-
 inline const char* StringBuffer::c_str() const noexcept
 {
+    static const char zero = 0;
```

Comment:
A `static const char` value will be allocated in read-only memory, no runtime checks needed. It's one of those rare cases where the compiler can actually use the `const` for something. 😛